### PR TITLE
Fix protobuf comparison

### PR DESCRIPTION
**Pull request by ilaleksin**
## Description

This PR addresses https://github.com/openfga/openfga/issues/894 by using **protocmp** functions to check that proto messages in unit tests are semantically equal

## References
https://github.com/openfga/openfga/issues/894

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Comment by ilaleksin** [comment]
Hey @jon-whit,

I prepared PR to fix https://github.com/openfga/openfga/issues/894

I see unit tests and linter jobs run only after a maintainer approval. Can you help me to run them?

**Comment by jon-whit** [comment]
> Hey @jon-whit,
> 
> I prepared PR to fix #894
> 
> I see unit tests and linter jobs run only after a maintainer approval. Can you help me to run them?

Yeah, no problem. Do you mind rebasing first please? Then I will approve them to run.

**Comment by ilaleksin** [comment]
> Yeah, no problem. Do you mind rebasing first please? Then I will approve them to run.

Hey @jon-whit, I have rebased the branch.
Please take a look again

**Review by jpadilla — APPROVED** [review]

**Pull request merged**
